### ISSUE TITLE
Add multi-threaded download to increase speed

### ISSUE THREAD

**Issue by mrglutton** [mauriciojost/olympus-photosync#30]
Could you add parallel image download so the transfer speed would be increased?

The current transfer speed on both G and N WiFi standards on OM-D E-M1 camera are limited to some 750 KiB/s, which is about 6 Mbit/s. On B standard it is about 100 KiB/s slower. The transfer graphs look capped, like the transfer is soft-limited; if the total transfer would be 5-8 concurrent files, and if the speed capping is done on per-file, basis (quite possible), that would meen some 40-50 Mbit/s speed on G and N networks.

Now, I couldn't test the camera on AC, but from connecting it to N wiFi adapter, it seems hardware inside is on N standard.

If the transfer would be stable, maybe it is possible to concurrently download 20-30 files at the same time, getting to the maximum of the N standard. 30X files DL at the same time would yield 20+ MiB/s.

**Comment by mauriciojost** [comment]
The camera behaves as mono-threaded when replying to HTTP requests (multiple parallel requests lock themselves). Have you tried downloading many files in parallel? Did you then observe faster downloads overall?
Implementing it is almost trivial, but would avoid making changes for no gain. I am charging my battery to test on my side. 
@mrglutton could you test on your EM1 (I own an EM10) too? Launch multiple instances of the app (on different directories) and see if download happens faster (even better if you can numerically measure it).

**Comment by mauriciojost** [comment]
Unfortunately for EM10 tests on mono-threaded download or multi-threaded downloads reach a maximum of 750KB. @mrglutton what about your EM1?

**Comment by mrglutton** [comment]
Tested it. Unfortunately, it seems there is really a cap on speed. 750 KiB/s, and no concurrent downloads. I tested this with multiple instances, but the camera serves stuff sequentially, one by one image. So... Tough luck. Maybe we could write to Olympus asking to implement this? :-) I know they would probably say no, but it would be such a phenomenal feature.

I really dislike taking the card out.
